src/error.rs

```diff
@@ -143,6 +146,24 @@ impl ErrorImpl {
         }
     }
 
+    /// Retrieve the object's associated backtrace, if any.
+    #[cfg(feature = "backtrace")]
+    fn backtrace(&self) -> Option<&Backtrace> {
+        match self {
+            #[cfg(feature = "dwarf")]
+            Self::Dwarf { backtrace, .. } => Some(backtrace),
+            Self::Io { backtrace, .. } => Some(backtrace),
+            Self::ContextOwned { .. } => None,
+            Self::ContextStatic { .. } => None,
+        }
+    }
+
+    /// Retrieve the object's associated backtrace, if any.
+    #[cfg(not(feature = "backtrace"))]
+    fn backtrace(&self) -> Option<&Backtrace> {
+        None
+    }
+
```

Comment:
Nice, that's basically what I was imagining when I was asking yesterday (apart from my later confusion about std::backtrace::Backtrace *type* availability).